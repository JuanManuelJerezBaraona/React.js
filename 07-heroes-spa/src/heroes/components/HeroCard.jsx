import { Link } from "react-router-dom";

export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters
}) => {

  const heroImageUrl = `/assets/heroes/${id}.jpg`;

  return (
    <div className="col-12 col-sm-6 col-md-4 col-lg-3">
      <div className="card">
        <img src={heroImageUrl} className="card-img" alt={superhero} />
        <div className="card-body">
          <h5 className="card-title">{superhero}</h5>
          <p className="card-text">{alter_ego}</p>

          {
            (alter_ego !== characters) && <p className="card-text">{characters}</p>
          }

          <p className="card-text">
            <small className="text-muted">{first_appearance}</small>
          </p>

          <Link to={`/hero/${id}`}>
            MÃ¡s...
          </Link>
        </div>
      </div>
    </div>
  )
}
