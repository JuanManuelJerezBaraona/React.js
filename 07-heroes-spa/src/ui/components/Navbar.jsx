import { Link, NavLink, useNavigate } from "react-router-dom";

export const Navbar = () => {

  const navigate = useNavigate();
  
  const onLogout = () => {
    navigate('/login', {
      replace: true
    });
  }

  return (
    <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
      <div className="container-fluid">
        <Link className="navbar-brand ms-2" to="/">Heroes</Link>
        <button className="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation">
          <span className="navbar-toggler-icon"></span>
        </button>
        <div className="collapse navbar-collapse" id="navbarNav">
          <ul className="navbar-nav text-center">

            <NavLink
              className={"nav-link ms-2"}
              to="/marvel"
            >
              Marvel
            </NavLink>

            <NavLink
              className={"nav-link ms-2"}
              to="/dc"
            >
              DC
            </NavLink>

            <span className="nav-link text-primary ms-2">
              Juan
            </span>

            <button 
              onClick={onLogout}
              className="nav-link btn">
              Logout
            </button>

          </ul>
        </div>
      </div>
    </nav>
  );
}
