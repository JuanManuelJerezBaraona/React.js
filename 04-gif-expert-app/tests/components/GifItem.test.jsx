import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";


describe('Test on <GifItem />', () => {

    const title = 'Death Note';
    const url = 'https://localhost/image.jpg';

    test('should match with snapshot', () => {

        const {container} = render( <GifItem title={title} url={url} /> );
        expect(container).toMatchSnapshot();

    });

    test('should show the image with the URL and ALT from the props', () => {

        render( <GifItem title={title} url={url} /> );
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);

    });

    test('should show the title in a paragraph', () => {

        render( <GifItem title={title} url={url} /> );
        expect(screen.getByText(title)).toBeTruthy();
        
    });

});