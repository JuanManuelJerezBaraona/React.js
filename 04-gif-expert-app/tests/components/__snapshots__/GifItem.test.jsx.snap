// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifItem /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Death Note"
      loading="lazy"
      src="https://localhost/image.jpg"
    />
    <p>
      Death Note
    </p>
  </div>
</div>
`;
